src/storage/src/frontend/mod.rs

```diff
@@ -183,7 +183,13 @@ impl<P: BackendStorage> FrontendStorage<P> {
 
         let mut to_write: Vec<Row> = vec![];
         let mut errors = HashMap::new();
+        let mut operation_error = None;
         for row in rows {
+            if row.len() > all_columns.len() {
+                operation_error = Some(OperationOnTableError::InsertTooManyExpressions);
+                break;
```

Comment:
I think you need to `return Ok(Err(OperationOnTableError::InsertTooManyExpressions))` otherwise you could select data out of table despite the error during an insertion.

Reply:
I agree. However, when I did it that way I was getting this error when the table or schema didn't exist (all_columns is empty). It might be better to check if the table or schema exist before processing the rows.

Reply:
Ah, I see. Seems like we need to introduce changes to backend storage ... My previous assumption that database can live without `"system.tables"` table (where list of tables is stored) does not hold anymore.
Can you please add `BackendStorage::is_table_exists(&self, namespace: &str, object_name: &str) -> bool` function and call it before `self.persistent.write`?
I will have a look how we can improve situation to work with the storage